File: easy-solution/minesp3164/700.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def searchBST(self, root, nval):
        while root:
            if root.val == nval:
                break
            elif root.val < nval:
                root = root.right
            else: root = root.left
        return root

        """
        :type root: TreeNode
        :type val: int
        :rtype: TreeNode
        """

File: easy-solution/galaxy4276/700.java
class Solution {
  public TreeNode searchBST(TreeNode root, int val) {
    if (root == null) return null;
    if (root.val == val) return root;
    TreeNode next = null;
    if (root.val > val) {
      if (root.left != null)
        next = searchBST(root.left, val);
    } else {
      if (root.right != null)
        next = searchBST(root.right, val);
    }

    return next;
  }
}